core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -171,7 +175,11 @@ class NewHadoopRDD[K, V](
 
       override def hasNext: Boolean = {
         if (!finished && !havePair) {
-          finished = !reader.nextKeyValue
+          try {
+            finished = !reader.nextKeyValue
+          } catch {
+            case _: EOFException if ignoreCorruptFiles => finished = true
```

Comment:
This is a behavior change to `NewHadoopRDD`, which may surprise the existing 1.6 users.